### PR TITLE
First attempt to add ability to send POST requests in promclient

### PR DESCRIPTION
**Pull request by mritunjaysharma394**
Signed-off-by: Mritunjay Sharma <mritunjaysharma394@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

In file `thanos-io/thanos/pkg/promclient/promclient.go`, I added a new function
that is trying to handle POST requests. 
This is the first draft, feedback appreciated to
get a better head-start.

This PR is the first attempt to handle https://github.com/thanos-io/thanos/issues/2577

## Verification

Not tested yet, it is the first draft to know if the right approach is being
followed or not and to seek more clarity on doubts.

### CODE REVIEW

File: pkg/promclient/promclient.go [hunk 1]
```diff
@@ -122,6 +122,36 @@ func (c *Client) get2xx(ctx context.Context, u *url.URL) (_ []byte, _ int, err e
 	return body, resp.StatusCode, nil
 }
 
+//First attempt to add ability to send POST requests [issue #2577]
```
Comment (kakkoyun) [inline pkg/promclient/promclient.go:125]: nit
```suggestion
// First attempt to add ability to send POST requests [issue #2577]
```

File: pkg/promclient/promclient.go [hunk 2]
```diff
@@ -122,6 +122,36 @@ func (c *Client) get2xx(ctx context.Context, u *url.URL) (_ []byte, _ int, err e
...
+// Doubt on which piece of code required to handle parameter/flag that
+// will give the option to choose between GET and POST
+func (c *Client) post2xx(ctx context.Context, u *url.URL, skipHead bool) (_ []byte, _ int, err error) {
```
Comment (kakkoyun) [inline pkg/promclient/promclient.go:128]: I'd love to see this method used in action. For example, I'd probably want to pass the content that I'd like to send.

**Review by kakkoyun — COMMENTED** [review]
Looking good. Thanks for doing it. 
I have left some comments.

Rather than proposing a code piece that hasn't been used it would be great to see it in action. You could add some test and example. But the ideally you could directly try to fix #2577 in a PR so that we see this code in action.

**Comment by GiedriusS** [comment]
Thanks for the review! I thought it should be easier - as simple as adding another parameter to the function which does the call to pass either GET or POST. I hope you don't mind if I take the implementation of this feature over.

**Pull request closed without merging**